effect/jvm/src/test/scala/scalaz/effect/generator/GenIO.scala

```diff
@@ -0,0 +1,145 @@
+package scalaz.effect.generator
+
+import org.scalacheck._
+import scalaz.effect.{ ExitResult, IO }
+
+trait GenIO {
+
+  /**
+   * Given a generator for `A`, produces a generator for `IO[E, A]` using the `IO.point` constructor.
+   */
+  def genSyncSuccess[E, A: Arbitrary]: Gen[IO[E, A]] =
+    for {
+      a <- Arbitrary.arbitrary[A]
+    } yield IO.point[E, A](a)
+
+  /**
+   * Given a generator for `A`, produces a generator for `IO[E, A]` using the `IO.async` constructor.
+   */
+  def genAsyncSuccess[E, A: Arbitrary]: Gen[IO[E, A]] =
+    for {
+      a <- Arbitrary.arbitrary[A]
+    } yield IO.async[E, A](cb => cb(ExitResult.Completed(a)))
+
+  /**
+   * Randomly uses either `genSyncSuccess`, or `genAsyncSuccess` with equal probability.
+   */
+  def genSuccess[E, A: Arbitrary]: Gen[IO[E, A]] = Gen.oneOf(genSyncSuccess[E, A], genAsyncSuccess[E, A])
+
+  /**
+   * Given a generator for `A`, produces a generator for `IO[E, A]` using the `IO.fail` constructor.
+   */
+  def genSyncFailure[E: Arbitrary, A]: Gen[IO[E, A]] =
+    for {
+      error <- Arbitrary.arbitrary[E]
+    } yield IO.fail(error)
+
+  /**
+   * Given a generator for `A`, produces a generator for `IO[E, A]` using the `IO.async` constructor.
+   */
+  def genAsyncFailure[E: Arbitrary, A]: Gen[IO[E, A]] =
+    for {
+      error <- Arbitrary.arbitrary[E]
+    } yield IO.async(cb => cb(ExitResult.Failed(error)))
+
+  /**
+   * Randomly uses either `genSyncFailure`, or `genAsyncFailure` with equal probability.
+   */
+  def genFailure[E: Arbitrary, A]: Gen[IO[E, A]] = Gen.oneOf(genSyncFailure[E, A], genAsyncFailure[E, A])
+
+  def genIO[E: Arbitrary: Cogen, A: Arbitrary: Cogen]: Gen[IO[E, A]] =
+    Gen.frequency(15 -> genSuccess[E, A], 1 -> genFailure[E, A])
+
+  /**
+   * Given a generator for `IO[E, A]`, produces a sized generator for `IO[E, A]` which represents a transformation,
+   * by using some random combination of the methods `map`, `flatMap`, `leftMap`, and any other method that does not change
+   * the success/failure of the value, but may change the value itself.
+   */
+  def genLikeTrans[E: Arbitrary: Cogen, A: Arbitrary: Cogen](implicit gen: Arbitrary[IO[E, A]]): Gen[IO[E, A]] =
```

Comment:
I'm not sure what ScalaCheck best practices are, but having this as an implicit parameter -- when this entire file shows how many different useful generation strategies for `IO` exist -- seems odd.